src/net.ts

```diff
@@ -30,29 +44,47 @@ export default class Network {
     });
   }
 
+  checkProtocolVersion(): Promise<any> {
+    return new Promise(async (resolve, reject) => {
+      const response = await this.provider.send('ain_checkProtocolVersion',
+          '', { version: this.protoVer });
+      if (response.code === 1) {
+        const nodeProtoVer = response.protoVer;
+        const VERSION_LIST = JSON.parse(fs.readFileSync(VERSIONS_PATH, 'utf-8'));
+        if (semver.lte(VERSION_LIST[SDK_VERSION].min, nodeProtoVer) &&
+              (!VERSION_LIST[SDK_VERSION].max ||
+                  semver.gte(VERSION_LIST[SDK_VERSION].max, nodeProtoVer))) {
+          // Update protoVer if we can
```

Comment:
Would it be helpful to emit some log messages?

Reply:
done